### ISSUE TITLE
Update the Trailer Size with the correct value which vaires with the protocol used

### ISSUE THREAD

**Issue by shrutigarg** [dotnet/corefx#4223]
This is to track the changes needed to update the Trailer size value in StreamSizes with the correct value

**Comment by vijaykota** [comment]
OpenSsl does not provide any macros or APIs to retrieve the header/trailer values. The values can change depending on the protocol and the ciphers being used in the low-level message exchanges.

Note that this has no functionality impact since the xplat code of System.Net.Security uses this only to compute the buffer size to be allocated for encrypted data. So a value higher than actual by a few bytes does not affect the logic. Also there are checks in place to guard against buffer overruns

**Comment by vijaykota** [comment]
Lowering the priority to "Future"

**Comment by bartonjs** [comment]
This is the root cause of RC2 escrow issue #7599.

**Comment by shrutigarg** [comment]
The way we have set the max trailer size to 68, we would need to do the same for header also. Since as said we don't have a way to retrieve the header/trailer values from libssl .We can set it the max value it can ever happen.. I had done some investigations during implementation around the header/trailer values that can happen in various protocols.. 
My findings were (when try to connect as windows client to Linux server)

SSL3:
header : 5
max : 16384

TLS:
header : 5
max : 16384

TLS11:
header : 21
max : 16384

TLS12:
header : 13
max : 16384

And this not only depends on the protocol but the cipher algorithms which are negotiated during connection .

We can increase the Header size value to 21( max among all) to get it working for all scenarios ..

And yes, we do have a [test case](https://github.com/dotnet/corefx/blob/master/src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs#L96) which tests write with large data (>16k bytes).
but seems it does not use "ECDHE-RSA-AES256-SHA384 " cipher which caused the failure .

And since there is no way to enforce the ciphers to be used during a connection through the any API or any other mean, It is not possible to add the exact testcase to repro this .
See the similar issue raised for cipher related tests : https://github.com/dotnet/corefx/issues/3832 .

BTW, I was able to repro the issue #7599 by running my server as 
"openssl s_server -cert mycert.pem -www -debug -cipher ECDHE-RSA-AES256-SHA384"

And writing the data >8000 bytes from a Linux client.. and saw the Assert happening. 
With Header size set to 21 ,, it was succeeding.
